wagtail_streamfield_migration_toolkit/utils.py

```diff
@@ -1,6 +1,253 @@
+            **kwargs
+        )
+
+    elif isinstance(block_def, ListBlock):
+        mapped_value = map_list_block_value(
+            block_value,
+            operation=operation,
+            block_def=block_def,
+            block_path=block_path,
+            **kwargs
+        )
+
+    elif isinstance(block_def, StructBlock):
+        mapped_value = map_struct_block_value(
+            block_value,
+            operation=operation,
+            block_def=block_def,
+            block_path=block_path,
+            **kwargs
+        )
+
+    else:
+        raise ValueError("Unexpected Structural Block + {}".format(block_value))
+
+    return mapped_value
+
+
+def map_stream_block_value(stream_block_value, block_def, block_path, **kwargs):
+    """
+    Maps each child block in a StreamBlock value.
+
+    Args:
+        stream_block_value:
+            The value of the StreamBlock, a list of child blocks
+        block_def:
+            The definition of the StreamBlock
+        block_path:
+            A '.' separated list of names of the blocks from the current block (not included) to
+            the nested block of which the value will be passed to the operation.
+
+    Returns
+        mapped_value:
+            The value of the StreamBlock after mapping all the children.
+    """
+
+    mapped_value = []
+    for child_block in stream_block_value:
+
+        # If the block is not at the start of `block_path`, then neither it nor its children are
+        # blocks that we need to change.
+        if not is_block_at_path_start(child_block["type"], block_path):
+            mapped_value.append(child_block)
+
+        else:
+            child_block_def = block_def.child_blocks[child_block["type"]]
+            mapped_child_value = map_block_value(
+                child_block["value"],
+                block_def=child_block_def,
+                block_path=block_path[1:],
+                **kwargs
+            )
+            mapped_value.append({**child_block, "value": mapped_child_value})
+
+    return mapped_value
+
+
+def map_struct_block_value(struct_block_value, block_def, block_path, **kwargs):
+    """
+    Maps each child block in a StructBlock value.
+
+    Args:
+        stream_block_value:
+            The value of the StructBlock, a dict of child blocks
+        block_def:
+            The definition of the StructBlock
+        block_path:
+            A '.' separated list of names of the blocks from the current block (not included) to
+            the nested block of which the value will be passed to the operation.
+
+    Returns
+        mapped_value:
+            The value of the StructBlock after mapping all the children.
+    """
+
+    mapped_value = {}
+    for (key, child_value) in struct_block_value.items():
+
+        # If the block is not at the start of `block_path`, then neither it nor its children are
+        # blocks that we need to change.
+        if not is_block_at_path_start(key, block_path):
```

Comment:
This comment is duplicated in the other location we call this function, maybe the function could have a better name? I would name it like `should_alter_block` or `should_alter_subtree`, and move the comment explaining "why" to the function itself.